boards/telosb/include/periph_conf.h

```diff
@@ -21,23 +21,27 @@
 #ifndef PERIPH_CONF_H
 #define PERIPH_CONF_H
 
+#include "macros/units.h"
+#include "periph_cpu.h"
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-/**
- * @name    Clock configuration
- * @{
- */
 #define CLOCK_CORECLOCK             (2457600U)
-#define MSP430_INITIAL_CPU_SPEED    2457600uL
-#define F_CPU                       MSP430_INITIAL_CPU_SPEED
-#define F_RC_OSCILLATOR             32768
-#define MSP430_HAS_DCOR             0
-#define MSP430_HAS_EXTERNAL_CRYSTAL 1
 
-#define CLOCK_CMCLK                 CLOCK_CORECLOCK     /* no divider programmed */
-/** @} */
+/**
+ * @brief   Clock configuration
+ */
+static const msp430_fxyz_clock_params_t clock_params = {
+    .dco_frequency = CLOCK_CORECLOCK,
+    .lfxt1_frequency = 32768,
+    .main_clock_source = MAIN_CLOCK_SOURCE_DCOCLK,
+    .submain_clock_source = SUBMAIN_CLOCK_SOURCE_DCOCLK,
```

Comment:
Note: The `MSP430_HAS_EXTERNAL_CRYSTAL` would indicate that the XT2 clock should be used instead of the DCO. However, the `msp430_init_dco()` in the TelosB's `board.c` configures and calibrates the DCO as clock source regardless of that macro. It seems to be a copy-paste error.

Also note: It should now be possible to run the DCO at 4 MHz instead of ~2.5 MHz, as the DCO calibration is a bit more robust.